src/main/java/org/zalando/nakadi/service/CursorsService.java

```diff
@@ -80,11 +81,37 @@ public CursorsService(final ZooKeeperHolder zkHolder,
 
         validateStreamId(cursors, streamId, subscriptionId);
 
+        LOG.debug("[COMMIT_CURSORS] stream IDs validation finished");
+
+        final Iterator<EventTypePartition> distinctPartitions = cursors.stream()
```

Comment:
What do you think about this approach?
1. group by etname_partition -> list of subscriptions
2. group by etname_partition -> max offset for the list of subscriptions